src/objects.ts

```diff
@@ -0,0 +1,35 @@
+export type validObjectValue<T> = Record<string, T> | null | undefined;
+
+export default function shallowEqualObjects<T>(
+  objA: validObjectValue<T>,
+  objB: validObjectValue<T>
```

Comment:
Same here. With these types I'd expect TypeScript to complain if you do `shallowEqualObjects({ a: 1 }, { a: true })`.

Reply:
Indeed. Thanks so much for catching. #33 for your review.